src/components/RadioGroup/RadioGroup.tsx

```diff
@@ -1,114 +1,148 @@
-    labelText,
-    helperText,
-    errorText,
-    showLabel,
-    errored = false,
-    required,
-    disabled,
-  } = props;
-  const baseName = "RadioGroup";
-  const footnote = errored ? errorText : helperText;
-  const direction = directionMap[layout];
-  const newChildren = [];
+    // Use Chakra's RadioGroup hook to set and get the proper props
+    // or the custom components.
+    const { getRootProps, getRadioProps } = useRadioGroup({
+      name,
+      defaultValue,
+      onChange,
+    });
+    const radioGroupProps = getRootProps();
 
-  React.Children.map(children, (child: React.ReactElement, i) => {
-    const attributes = { key: i };
-    if (child !== undefined && child !== null) {
-      attributes["name"] = name;
-      if (disabled) {
-        attributes["disabled"] = true;
-        attributes["aria-disabled"] = true;
-      }
-      if (errored) {
-        attributes["errored"] = true;
-      }
-      if (child.props.value === defaultValue) {
-        attributes["checked"] = true;
+    // Go through the Radio children and update them as needed.
+    React.Children.map(children, (child: React.ReactElement, i) => {
+      const chakraRadioProps = getRadioProps({
+        value: child.props.value,
+      } as any);
+      const newProps = { key: i };
+
+      if (child !== undefined && child !== null) {
+        if (isDisabled) {
+          newProps["isDisabled"] = true;
+        }
+        if (isInvalid) {
+          newProps["isInvalid"] = true;
+        }
+        if (isRequired) {
+          newProps["isRequired"] = true;
+        }
+        if (child.props.value === defaultValue) {
+          newProps["checked"] = true;
+        }
+        newChildren.push(
+          React.cloneElement(child, { ...newProps, ...chakraRadioProps })
+        );
       }
-      newChildren.push(React.cloneElement(child, { ...attributes }));
-    }
-  });
+    });
 
-  return (
-    <fieldset id={`radio-group-${id}`} className={`${baseName} ${className}`}>
-      <legend className={showLabel ? "" : "sr-only"}>
-        <span>{labelText}</span>
-        <span>{required ? "required" : "optional"}</span>
-      </legend>
-      <ChakraRadioGroup
-        name={name}
-        defaultValue={defaultValue}
-        onChange={onChange}
+    // Get the Chakra-based styles for all the custom elements in this component.
+    const styles = useMultiStyleConfig("CustomRadioGroup", {});
+
+    return (
+      <Box
+        as="fieldset"
+        id={`radio-group-${id}`}
+        className={className}
+        __css={styles}
       >
-        <Stack direction={[direction]} spacing={spacing.s}>
+        <legend className={showLabel ? "" : "sr-only"}>
+          <span>{labelText}</span>
+          {optReqFlag && (
+            <Box as="span" __css={styles.required}>
+              {isRequired ? "Required" : "Optional"}
+            </Box>
+          )}
+        </legend>
+        <Stack
+          direction={[direction]}
+          spacing={spacing.s}
```

Comment:
When `direction` is set to `row`, the spacing should be `2rem`, which aligns with `spacing.l`.